### PR TITLE
refactor argmin/max to exprmin/max and make it calcite compliant

### PR DESCRIPTION
**Pull request by jasperjiaguo**
First PR for https://github.com/apache/pinot/issues/11254
Rename argmin/max function to exprmin/max, swap the measuring/projection column

new signature:
exprmin/max(projectCol, measureCol1 [, measureCol2]*)

### CODE REVIEW

**Review by xiangfu0 — APPROVED** [review]

File: pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/AggregationFunctionType.java [hunk 1]
```diff
@@ -254,12 +254,12 @@ public enum AggregationFunctionType {
 
   // argMin and argMax
   // TODO: argmin/argmax syntax not conformed with Calcite. we need to migrate.
-  ARGMIN("argMin"),
-  ARGMAX("argMax"),
-  PARENTARGMIN(CommonConstants.RewriterConstants.PARENT_AGGREGATION_NAME_PREFIX + ARGMIN.getName()),
-  PARENTARGMAX(CommonConstants.RewriterConstants.PARENT_AGGREGATION_NAME_PREFIX + ARGMAX.getName()),
-  CHILDARGMIN(CommonConstants.RewriterConstants.CHILD_AGGREGATION_NAME_PREFIX + ARGMIN.getName()),
-  CHILDARGMAX(CommonConstants.RewriterConstants.CHILD_AGGREGATION_NAME_PREFIX + ARGMAX.getName()),
+  EXPRMIN("exprMin"),
```
Comment (xiangfu0) [inline pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/AggregationFunctionType.java:257]: You may need to update here for making v2 query engine work

Comment (jasperjiaguo) [reply pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/AggregationFunctionType.java]: Updated, PTAL, thanks!

**Comment by siddharthteotia** [comment]
@jasperjiaguo  - can we update the user docs please when you get a chance ?

**Comment by jasperjiaguo** [comment]
> @jasperjiaguo - can we update the user docs please when you get a chance ?

fixed https://docs.pinot.apache.org/configuration-reference/functions/arg_min-arg_max

**Pull request merged**